eng/common/testproxy/test-proxy.ps1

```diff
@@ -0,0 +1,162 @@
+Set-StrictMode -Version 4
+$AVAILABLE_TEST_PROXY_BINARIES = @{
+    "Windows" = @{
+        "AMD64" = @{
+            "system" = "Windows"
+            "machine" = "AMD64"
+            "file_name" = "test-proxy-standalone-win-x64.zip"
+            "executable" = "Azure.Sdk.Tools.TestProxy.exe"
+        }
+    }
+    "Linux" = @{
+        "X86_64" = @{
+            "system" = "Linux"
+            "machine" = "X86_64"
+            "file_name" = "test-proxy-standalone-linux-x64.tar.gz"
+            "executable" = "Azure.Sdk.Tools.TestProxy"
+        }
+        "ARM64" = @{
+            "system" = "Linux"
+            "machine" = "ARM64"
+            "file_name" = "test-proxy-standalone-linux-arm64.tar.gz"
+            "executable" = "Azure.Sdk.Tools.TestProxy"
+        }
+    }
+    "Darwin" = @{
+        "X86_64" = @{
+            "system" = "Darwin"
+            "machine" = "X86_64"
+            "file_name" = "test-proxy-standalone-osx-x64.zip"
+            "executable" = "Azure.Sdk.Tools.TestProxy"
+        }
+        "ARM64" = @{
+            "system" = "Darwin"
+            "machine" = "ARM64"
+            "file_name" = "test-proxy-standalone-osx-arm64.zip"
+            "executable" = "Azure.Sdk.Tools.TestProxy"
+        }
+    }
+}
+
+function Get-SystemArchitecture {
+    $unameOutput = uname -m
+    switch ($unameOutput) {
+        "x86_64" { return "X86_64" }
+        "aarch64" { return "ARM64" }
+        "arm64" { return "ARM64" }
+        default { throw "Unable to determine system architecture. uname -m returned $unameOutput." }
+    }
+}
+
+function Get-Proxy-Meta () {
+    $ErrorActionPreferenceDefault = $ErrorActionPreference
+    $ErrorActionPreference = "Stop"
+
+    $os = "unknown"
+    $machine = Get-SystemArchitecture
```

Comment:
Won't this halt/throw on a non-ubuntu OS? Should you put this check within the `$IsLinux` conditional instead?

Reply:
Nope! Works just fine 👍 

Here's an [example](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3929950&view=logs&j=2993fb17-9fe5-5b0b-e997-efa0cab8f900&t=cf38eb1d-d0e9-5773-1107-71d89ce67904&l=21).